[NEXUS-4530] test repo keyword search failure

adreghiciu opened the pull request.

adreghiciu committed:
List nexus-configuration-model as a direct dependency

nexus-configuration-model is directly used so it should be a direct dependency not a transitive

Signed-off-by: Alin Dreghiciu <adreghiciu@gmail.com>

adreghiciu committed:
Merge branch 'refs/heads/master' into nexus-4530-testRepoKeywordSearch-failure

adreghiciu committed:
[NEXUS-4530] Wait for repair tasks to finish before running tests

Intermittent failure of DefaultIndexerManagerTest#testRepoKeywordSearch was caused by the fact that before running the test, remote url of "apache-snapshots" repository was updated. This triggered an reindex task to repair the index of apache snapshots. The reindex task runs at undefined pint in time. In case that the reindex task was started and did not finished before the test was run, the reindexing started by the test will simply bypass the indexing of apache snapshots as it is considered that reindexing will anyhow be performed. when this happen the search results will not contain artifacts from apache snapshots, so the test will fail.

Signed-off-by: Alin Dreghiciu <adreghiciu@gmail.com>

velo commented on nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java:
```diff
@@ -140,6 +140,11 @@ public abstract class AbstractMavenRepoContentTests
+        // wait for this tasks to run
+        // waiting was introduced as a fix for NEXUS-4530 but will benefit all subclasses of this test as it will
+        // avoid unexpected behavior
+        waitForTasksToStop();
```
Just checking, but shouldn't you events.waitForQuietPeriod() as well?

adreghiciu replied:
That part is not involved in issue in cause but that will not hurt. I will add it.

adreghiciu replied:
Implemented in 4260e9a2

adreghiciu committed:
[NEXUS-4530] Wait also for async events to stop

This is not directly related but it will ensure that everything is calm before proceeding.
We may consider actually running all indexing related jobs in SYNC mode as ITs are doing.

Signed-off-by: Alin Dreghiciu <adreghiciu@gmail.com>

velo commented:
+1

peterlynch commented on nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java:
```diff
@@ -140,6 +140,12 @@ public abstract class AbstractMavenRepoContentTests
+        // avoid unexpected behavior
+        wairForAsyncEventsToCalmDown();
+        waitForTasksToStop();
     }
```
We should get in the habit of adding missing javadoc as we touch old files, especially on public methods such as fillInRepo(). Would you mind adding some that summarize what is going on in this method? method name fillInRepo tells me nothing....

nabcos commented:
+1

cstamas commented:
+1
